src/PrestaShopBundle/Entity/Repository/LogRepository.php

```diff
@@ -252,12 +252,18 @@ private function buildGridQuery(SearchCriteriaInterface $searchCriteria)
             }
 
             if ('date_add' == $filterName) {
-                if (!empty($filterValue['from']) &&
-                    !empty($filterValue['to'])
-                ) {
-                    $qb->andWhere('l.date_add BETWEEN :date_from AND :date_to');
-                    $qb->setParameter('date_from', $filterValue['from']);
-                    $qb->setParameter('date_to', $filterValue['to']);
+                $qb->andWhere('l.date_add >= :date_from AND l.date_add <= :date_to');
+                $qb->setParameter('date_from', sprintf('%s 0:0:0', $filterValue['from']));
+                $qb->setParameter('date_to', sprintf('%s 23:59:59', $filterValue['to']));
+
+                if (isset($filterValue['from'])) {
+                    $qb->andWhere('l.date_add >= :date_from');
```

Comment:
I don't understand why you duplicate this condition from line 255.

You must choice between using:
```php
                $qb->andWhere('l.date_add >= :date_from AND l.date_add <= :date_to');
                $qb->setParameter('date_from', sprintf('%s 0:0:0', $filterValue['from']));
                $qb->setParameter('date_to', sprintf('%s 23:59:59', $filterValue['to']));
```

and 
```php
                if (isset($filterValue['from'])) {
                    $qb->andWhere('l.date_add >= :date_from');
                    $qb->setParameter('date_from', sprintf('%s 0:0:0', $filterValue['from']));
                }

                if (isset($filterValue['to'])) {
                    $qb->andWhere('l.date_add <= :date_to');
                    $qb->setParameter('date_to', sprintf('%s 23:59:59', $filterValue['to']));
                }
```

But you can't use both, it's useless 🤔

Reply:
I have done a new commit now. The reason I inserted the condition twice was because I found the same condition inside the file CustomerQueryBuilder.php, in src/Core/Grid/Query. See [here](https://github.com/PrestaShop/PrestaShop/blob/30b604fbcfaffed5e018ba15a82b11e53cb9a58f/src/Core/Grid/Query/CustomerQueryBuilder.php#L211-L227)

Reply:
> I have done a new commit now. The reason I inserted the condition twice was because I found the same condition inside the file CustomerQueryBuilder.php, in src/Core/Grid/Query. See [here](https://github.com/PrestaShop/PrestaShop/blob/30b604fbcfaffed5e018ba15a82b11e53cb9a58f/src/Core/Grid/Query/CustomerQueryBuilder.php#L211-L227)

It was good to reuse another part of code but I think this part of code actually is buggy 😄 I'll check it